waku/v2/rpc/relay.go

```diff
@@ -44,6 +72,7 @@ func (r *RelayService) PostV1Subscription(req *http.Request, args *TopicsArgs, r
 			reply.Error = err.Error()
 			return nil
 		}
+		r.messages[topic] = make([]*pb.WakuMessage, 0)
```

Comment:
Shouldn't this happen before the `.Publish()`?, also lock r.messages
```suggestion
                r.messagesMutex.Lock()
                defer r.messagesMutex.Unlock()
		r.messages[topic] = make([]*pb.WakuMessage, 0)
```

Reply:
What if publish fail? I am adding at the point I am sure that the call will be successful if that makes sense